app/models/user.rb

```diff
@@ -286,7 +288,18 @@ def assign_barcode!
        'user' => {email: self.email}
       })
 
-    last_user_barcode = User.where('barcode < 27777099999999').order(:barcode).pluck(:barcode).last
+    # Integer(string) can raise ArgumentError.  we choose not to rescue it, because
+    # not having min and max barcode range boundaries set in the application.yml file
+    # should prevent the app from working in a visible manner.
+    min_barcode = Integer(ENV['USER_BARCODE_ALLOTTED_RANGE_MINIMUM'])
+    max_barcode = Integer(ENV['USER_BARCODE_ALLOTTED_RANGE_MAXIMUM'])
+
+    last_user_barcode = User.where("barcode < #{max_barcode}").order(:barcode).pluck(:barcode).last
+    # no non-nil barcodes found?  this should never happen, but let's make sure we can handle it
+    if (last_user_barcode.blank? || (last_user_barcode < min_barcode))
+      last_user_barcode = min_barcode
+    end
+
```

Comment:
Hi Darya,
I like the use of ENVs — that's definitely clearer. A few minor points:

- it's a bit more Rubyish to use `.to_i` rather than `Integer()`, but not a big deal
- I think you want `.nil?` not `.blank?` here
- One way of getting a `nil` is if there are no barcodes at all; another though, is if we've run out of barcodes, and I think many developers reading this would assume that was the most likely case (I did, for a sample of 1!) So I would either check for three error cases — no barcodes at all; barcode less than min_barcode; run out of barcodes — and handle each separately; or, at the very least explain better in comments what is happening.

Reply:
Well, I think blank will cover a nil.  I was using this as a guide:  https://blog.appsignal.com/2018/09/11/differences-between-nil-empty-blank-and-present.html .

Reply:
Sure, I'll add a max range check and more docs.

Reply:
> Well, I think blank will cover a nil.
Ah, gotcha. Did not realise this was a Rails addition. Thanks!

Reply:
Looks good — thanks!